src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -643,7 +647,10 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
                     // At this point, only the amounts on each allocation is correct, not the percentage.
                     // Go through the allocation and adjust the percentage so it is correct.
                     for (Allocation__c consolidatedAllocation : consolidatedAllocationList) {
-                        consolidatedAllocation.Percent__c = consolidatedAllocation.Amount__c / amount * 100;
+                        // Only recalculate if percent values were not set on the Allocations
+                        if(consolidatedAllocation.Percent__c == null){
+                            consolidatedAllocation.Percent__c = consolidatedAllocation.Amount__c / amount * 100;
```

Comment:
Since this is about fixing rounding issues, is there any value to implementing the `Divide()` method so that you can explicitly specify scale and the rounding method? The `PMT_PaymentWizard_CTRL` class uses the `Divide` method as well. For example:
```
Decimal consolidatedAmt = consolidatedAllocation.Amount__c;
Decimal consolidatedPercent = consolidatedAmt.divide(amount, 2, System.RoundingMode.HALF_UP);
consolidatedAllocation.Percent__c = consolidatedPercent;
```

https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_decimal.htm

Reply:
I like this idea, since it gives more control to the rounding, but I'm still not convinced we **want** to round these percents, given the possibility that a user could have entered a long decimal, for whatever reason.

Reply:
Looking at this again, this code only gets called if the user didn't enter a Percent value for all of their Allocations, **but** rounding the percent value could technically make it less accurate than the amount value...

Reply:
Understood. Floating point fields are always a bit challenging to understand for me. Even though we define a numeric field with 2 decimal points (scale), Salesforce actually stores the data with (what I think is) nearly unlimited number of values after the decimal point. From what I've seen, the scale is only used for rounding and displaying the value in the UI. So in this case I'm pretty sure we're actually storing the full result of the calculation as well as using it for the resulting calculations. And, that is probably better for this use case than forcing the rounding and having to deal with what could be a fractional remainder.